File: Exercism_Solutiuon/Scala/Bob.scala
object Bob {

  def hasLetters(word: String): Boolean ={
    return(word.exists(_.isLetter))
  }

  def silence(word: String) : Boolean = {
    return(word.forall(_.isWhitespace))
  }

  def shouting(word: String) : Boolean = {
    return(word.toUpperCase().equals(word))
  }

  def question(word: String): Boolean = {
    val wordTrimed= word.trim()
    return(wordTrimed.takeRight(1) == "?")
  }

  
  def response(statement: String): String = {
    if(shouting(statement) && hasLetters(statement) && question(statement)){
      return("Calm down, I know what I'm doing!");
    } else if(question(statement)){
      return("Sure.");
    } else if(shouting(statement) && question(statement) || silence(statement)){
      return("Fine. Be that way!");
    }else if(shouting(statement) && hasLetters(statement)){
      return("Whoa, chill out!");
    } else {
      return("Whatever.");
    }
  }  
}

File: Exercism_Solutiuon/R/bob.R
bob <- function(input) {
  # Remove leading and trailing whitespaces
  input <- trimws(input)
  
  if (nchar(input) == 0) {
    return("Fine. Be that way!")
  } else if (grepl("[A-Z]", input) && toupper(input) == input) {
    if (substr(input, nchar(input), nchar(input)) == "?") {
      return("Calm down, I know what I'm doing!")
    } else {
      return("Whoa, chill out!")
    }
  } else if (substr(input, nchar(input), nchar(input)) == "?") {
    return("Sure.")
  } else {
    return("Whatever.")
  }
}


#################################################################################

input <- "Does this cryogenic chamber make me look fat?"

bob(input)


substr(input, nchar(input), nchar(input))


my_string <- "Hello, World!"

# Get the last character
last_character <- substr(my_string, nchar(my_string), nchar(my_string))

# Print the result
print(last_character)
